### PR TITLE
Add /api/v0/config/show endpoint

### PR DESCRIPTION
**Pull request by xicombd**
Implements the `/api/v0/config/show` as specified at https://github.com/ipfs/http-api-spec/pull/29

Related to #49 & very similar to #72

### CODE REVIEW

File: tests/test-http-api/test-config.js [hunk 1]
```diff
@@ -221,5 +234,15 @@ describe('config', () => {
         })
       })
     })
+
+    describe('ipfs.config.show', () => {
```
Comment (diasdavid) [inline tests/test-http-api/test-config.js]: why a describe here?

**Comment by diasdavid** [comment]
overall LGTM :) Let's just take out the unnecessary `describe`

File: tests/test-http-api/test-config.js (hunk 1 above)
Comment (xicombd) [inline tests/test-http-api/test-config.js]: Was just trying to keep the structure of the other endpoints' tests, removed both of them

**Comment by diasdavid** [comment]
woot! :D

**Pull request merged**